### PR TITLE
Code refactor and optimize r2_gi feature

### PR DESCRIPTION
**Pull request by GermanAizek**
@Xottab-DUTY, Improving code readability and performance

### CODE REVIEW

File: src/Layers/xrRender/light.cpp [hunk 1]
```diff
@@ -208,11 +208,12 @@ void light::spatial_move()
     // update spatial DB
     SpatialBase::spatial_move();
 
-#if (RENDER == R_R2) || (RENDER == R_R3) || (RENDER == R_R4)
-    if (flags.bActive)
+#if (RENDER == R_GL) || (RENDER == R_R2) || (RENDER == R_R3) || (RENDER == R_R4)
+    if (flags.bActive) {
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/light.cpp]: Please, fix the code style: brace should always be on a new line.
```suggestion
    if (flags.bActive)
    {
```

Comment (GermanAizek) [reply src/Layers/xrRender/light.cpp]: @Xottab-DUTY fix here https://github.com/OpenXRay/xray-16/pull/1148/commits/a6e64361ad518397516a76a5f9bd99481aa3c879

Comment (Xottab-DUTY) [reply src/Layers/xrRender/light.cpp]: Don't forget about other braces in this pull request – they should be fixed too ;)

Comment (Xottab-DUTY) [reply src/Layers/xrRender/light.cpp]: So...?

File: src/Layers/xrRender/light_gi.cpp [hunk 2]
```diff
@@ -21,8 +21,12 @@ void light::gi_generate()
         switch (flags.type)
         {
         case IRender_Light::POINT: dir.random_dir(random); break;
-        case IRender_Light::SPOT: dir.random_dir(direction, cone, random); break;
-        case IRender_Light::OMNIPART: dir.random_dir(direction, cone, random); break;
+        case IRender_Light::DIRECT:
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/light_gi.cpp:24]: This changes behaviour! Please, keep pull request to it's topic.
This pull request says "optimize and refactor". Not adding new features. Please, remove new features and submit as separate pull request!

File: src/Layers/xrRender_R2/r2_rendertarget_accum_reflected.cpp [hunk 3]
```diff
@@ -67,7 +42,9 @@ void CRenderTarget::accum_reflected(light* L)
         RCache.set_Shader(s_accum_reflected);
 
         // Constants
-        RCache.set_c("Ldynamic_pos", L_pos.x, L_pos.y, L_pos.z, 1 / (L->range * L->range));
+        float att_R = L->range * .95f;
+        float att_factor = 1.f / (att_R * att_R);
+        RCache.set_c("Ldynamic_pos", L_pos.x, L_pos.y, L_pos.z, att_factor);
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender_R2/r2_rendertarget_accum_reflected.cpp:47]: This changes original behaviour. Please, remove this change and submit in a separate pull request along with direct light sources.

**Comment by Xottab-DUTY** [comment]
@GermanAizek, please, update this PR according to the review comments

**Comment by Xottab-DUTY** [comment]
@GermanAizek

**Comment by Xottab-DUTY** [comment]
Thanks for the pull request and highlighting optimizations and a bug fix!
